File: android/networking/src/main/kotlin/dev/example/moodlenative/networking/LmsModuleCompletionSummary.kt
package dev.example.moodlenative.networking

data class LmsModuleCompletionSummary(
    val progress: Double,
    val completedRequirementCount: Int,
    val totalRequirementCount: Int,
    val isComplete: Boolean,
    val accessibilityLabel: String,
) {
    val isInProgress: Boolean
        get() = !isComplete && progress > 0.0
}

File: ios/Shared/Sources/Networking/LmsModuleCompletionSummary.swift
import Foundation

public struct LmsModuleCompletionSummary: Equatable, Sendable {
    public let progress: Double
    public let completedRequirementCount: Int
    public let totalRequirementCount: Int
    public let isComplete: Bool
    public let accessibilityLabel: String

    public var isInProgress: Bool {
        isComplete == false && progress > 0
    }
}
